scalameta/testkit/src/main/scala/scala/meta/testkit/StringFS.scala

```diff
@@ -0,0 +1,94 @@
+package scala.meta.testkit
+
+import scala.meta.io.AbsolutePath
+import java.io.File
+import java.nio.charset.StandardCharsets
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.StandardOpenOption
+import scala.meta.internal.io.FileIO
+import scala.meta.io.RelativePath
+
+object StringFS {
+
+  /**
+    * Creates a temporary directory with a layout matching the markup in the string.
+    *
+    * Example syntax of the expected markup in the string:
+    * {{{
+    *   fromString("""
+    *   /build.sbt
+    *   lazy val core = project
+    *   /src/main/scala/core/Foo.scala
+    *   package core
+    *   object Foo.scala
```

Comment:
Fancy!

Reply:
I already use `StringFS` in scalafmt and scalafix for a long time, it's very helpful!